pkg/prometheus/operator.go

```diff
@@ -106,3 +121,95 @@ func ValidateRemoteWriteSpec(spec monitoringv1.RemoteWriteSpec) error {
 
 	return nil
 }
+
+// Process is a helper function that retrieves the status subresource of the object identified by the given
+// key.
```

Comment:
nit: WDYT like this?
```suggestion
// Process will determine the Status of a Prometheus resource (server or agent) depending on its current state in the cluster
```